frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts

```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
+          },
+          () => this.logger.info('EDIT-TERMS-AND-CONDITIONS-FINISHED')
+        );
+    };
+    
+    /**
+     * Parameters of the modal
+     */
+    const PARAMS = {
+      title: 'Edit Challenge Terms and Descriptions',
+      content: '',
+      confirm: 'Submit',
+      deny: 'Cancel',
+      form: [
+        {
+          name: 'Challenge Terms and Descriptions',
+          isRequired: true,
+          label: 'terms_and_conditions_file',
+          placeholder: '',
+          type: 'file',
+          value: '',
+        },
+      ],
+      confirmCallback: SELF.apiCall,
+    };
+    SELF.globalService.showModal(PARAMS);
+  }
+
+  /**
+   * Edit challenge evaluation criteria with file function
+   */ 
+  editEvaluationCriteriaUpload() {
+    const SELF = this;
+    SELF.apiCall = (params) => {
+      const FORM_DATA: FormData = new FormData();
+      FORM_DATA.append('evaluation_criteria_file', params['evaluation_criteria_file']);
+      SELF.apiService
+        .patchFileUrl(
+          SELF.endpointsService.editChallengeDetailsURL(SELF.challenge.creator.id, SELF.challenge.id),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            SELF.challenge.evaluation_details = data.evaluation_details;
+            SELF.globalService.showToast('success', 'The evaluation details is successfully updated!', 5);
+          },
+          (err) => {
+            SELF.globalService.handleApiError(err, true);
+            SELF.globalService.showToast('error', err);
+          },
+          () => this.logger.info('EDIT-CHALLENGE-EVALUATION-DETAILS-FINISHED')
+        );
+    };
+
+    /**
+     * Parameters of the modal
+     */
+    const PARAMS = {
+      title: 'Edit Challenge Evaluation Criteria',
+      content: '',
+      confirm: 'Submit',
+      deny: 'Cancel',
+      form: [
+        {
+          name: 'Challenge Evaluation Criteria',
+          isRequired: true,
+          label: 'evaluation_criteria_file',
+          placeholder: '',
+          type: 'file',
+          value: '',
+        },
+      ],
+      confirmCallback: SELF.apiCall,
+    };
+    SELF.globalService.showModal(PARAMS);
+ 
+  }
+
+  /**
+   * Edit phase details criteria with file function
+   */ 
+  editPhaseDetailsUpload() {
+    const SELF = this;
+    SELF.apiCall = (params) => {
+      const FORM_DATA: FormData = new FormData();
+      FORM_DATA.append('phase_description_file', params['phase_description_file']);
+      SELF.apiService
+        .patchFileUrl(
+          SELF.endpointsService.updateChallengePhaseDetailsURL(SELF.challenge.id, SELF.selectedPhase['id']),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            for (const attrname of Object.keys(data)) {
+              SELF.selectedPhase[attrname] = data[attrname];
+            }
+            SELF.globalService.showToast('success', 'The challenge phase description is successfully updated!');
```

Comment:
Please change the message to `Challenge phase description updated successfully!`